src/Oro/Bundle/EntityExtendBundle/Extend/EntityProcessor.php

```diff
@@ -85,6 +93,9 @@ public function updateDatabase($warmUpConfigCache = false, $updateRouting = fals
             );
         }
 
+        $event = new ExtendSchemaUpdateEvent($updateRouting);
+        $this->eventDispatcher->dispatch(ExtendSchemaUpdateEvent::NAME, $event);
```

Comment:
Actually looks good as an idea to add events before and after commands execution, but in this case the events name 'extend_schema.successful_update' is confusing a bit because a real update will be later - after executeCommand.
Also it will be better to pass commands into event instead of "$updateRouting", in this case it will be more useful.

So please move it after "executeCommands" and if it returns TRUE it will mean "succesful_update", and rename events to e.g. "extend_schema.update_database.after" and add the similar one "extend_schema.update_database.before"

And if you do really need to clear navigation cache before update do it in "...before" event.